src/OctonetData.cpp

```diff
@@ -287,11 +301,53 @@ PVR_ERROR OctonetData::GetChannels(bool radio, kodi::addon::PVRChannelsResultSet
   return PVR_ERROR_NO_ERROR;
 }
 
+PVR_ERROR OctonetData::GetChannelStreamProperties(const kodi::addon::PVRChannel& channelinfo, std::vector<kodi::addon::PVRStreamProperty>& properties)
+{
+  std::string openUsingDescription;
+
+  if (m_openStreamUsing != OpenStreamUsing::LIVE_STREAM_API)
+  {
+    switch (m_openStreamUsing)
+    {
+      case OpenStreamUsing::LIVE_STREAM_API:
+        openUsingDescription = kodi::addon::GetLocalizedString(30003); // Add-on default handling
+        break;
+      case OpenStreamUsing::KODI_INPUTSTREAM:
+        openUsingDescription = kodi::addon::GetLocalizedString(30004); // Kodi directly
+        break;
+      case OpenStreamUsing::INPUTSTREAM_FFMPEGDIRECT:
+        openUsingDescription = kodi::addon::GetLocalizedString(30005); // inpustream.ffmpegdirect;
+        properties.emplace_back(PVR_STREAM_PROPERTY_INPUTSTREAM, "inputstream.ffmpegdirect");
+        properties.emplace_back("inputstream.ffmpegdirect.is_realtime_stream", "true");
+        properties.emplace_back("inputstream.ffmpegdirect.open_mode", "ffmpeg");
+        // This property is required to supoprt timeshifting for Radio channels
+        properties.emplace_back("inputstream-player", "videodefaultplayer");
+        properties.emplace_back("inputstream.ffmpegdirect.stream_mode", "timeshift");
+        break;
+      case OpenStreamUsing::INPUTSTREAM_RTMP:
+        openUsingDescription = kodi::addon::GetLocalizedString(30006); // inpustream.rtmp;
+        properties.emplace_back(PVR_STREAM_PROPERTY_INPUTSTREAM, "inputstream.rtmp");
+        break;
+      default:
+        break;
```

Comment:
If you end up here, is this an error condition (return PVR_ERROR_...)?
At least the following log will have an empty `openUsingDescription`, which seems not intended.